src/plugins/discover/public/application/apps/main/discover_main_route.tsx

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useEffect, useState, memo } from 'react';
+import { History } from 'history';
+import { useParams } from 'react-router-dom';
+import { SavedObject as SavedObjectDeprecated } from 'src/plugins/saved_objects/target/types/public';
```

Comment:
@majagrubic This should fix the type check (it did in my test PR), I bet it was Webstorm who suggested that import :)
```suggestion
import type { SavedObject as SavedObjectDeprecated } from '../../../../../saved_objects/public';
```